pennylane/ops/op_math/controlled_class.py

```diff
@@ -0,0 +1,380 @@
+    def __init__(
+        self, base, control_wires, control_values=None, work_wires=None, do_queue=True, id=None
+    ):
+        control_wires = Wires(control_wires)
+        work_wires = Wires([]) if work_wires is None else Wires(work_wires)
+
+        if control_values is None:
+            control_values = [True] * len(control_wires)
+        else:
+            if isinstance(control_values, str):
+                warnings.warn(
+                    "Specifying control values as a string is deprecated. Please use Sequence[Bool]",
+                    UserWarning,
+                )
+                control_values = [(x == "1") for x in control_values]
+
+            if len(control_values) != len(control_wires):
+                raise ValueError("control_values should be the same length as control_wires")
+            if not set(control_values).issubset({False, True}):
+                raise ValueError("control_values can only take on True or False")
+
+        if len(Wires.shared_wires([base.wires, control_wires])) != 0:
+            raise ValueError("The control wires must be different from the base operation wires.")
+
+        if len(Wires.shared_wires([work_wires, base.wires + control_wires])) != 0:
+            raise ValueError(
+                "Work wires must be different the control_wires and base operation wires."
+            )
+
+        self.hyperparameters["control_wires"] = control_wires
+        self.hyperparameters["control_values"] = control_values
+        self.hyperparameters["work_wires"] = work_wires
+
+        self._name = f"C{base.name}"
+
+        super().__init__(base, do_queue, id)
+
+    # Properties on the control values ######################
+    @property
+    def control_values(self):
+        """Iterable[Bool]. For each control wire, denotes whether to control on ``True`` or
+        ``False``."""
+        return self.hyperparameters["control_values"]
+
+    @property
+    def _control_int(self):
+        """Int. Conversion of ``control_values`` to an integer."""
+        return sum(2**i for i, val in enumerate(reversed(self.control_values)) if val)
+
+    # Properties on the wires ##########################
+
+    @property
+    def control_wires(self):
+        """The control wires."""
+        return self.hyperparameters["control_wires"]
+
+    @property
+    def target_wires(self):
+        """The wires of the target operator."""
+        return self.base.wires
+
+    @property
+    def work_wires(self):
+        """Additional wires that can be used in the decomposition. Not modified by the operation."""
+        return self.hyperparameters["work_wires"]
+
+    @property
+    def wires(self):
+        return self.control_wires + self.base.wires + self.work_wires
+
+    # pylint: disable=protected-access
+    @property
+    def _wires(self):
+        return self.wires
+
+    # pylint: disable=protected-access
+    @_wires.setter
+    def _wires(self, new_wires):
```

Comment:
maybe add a comment or doc string mentioning the order in which the wires need to be passed (either here or in the `wires` property?

Reply:
The order is the same as the order of `wires`. I'm not quite sure I see the confusion.

Especially as setting `_wires` this way is something that "shouldn't" be done, but is.  We want to hide this away and discourage it.

Reply:
yea probably not best to have it in the private method, but I don't think there is a comment or docstring  in the wires property that mentions this ordering either